w2c2/c.c

```diff
@@ -1031,15 +1046,17 @@ wasmCWriteLiteral(
         case wasmValueTypeF32: {
             U32 bits = (U32) value.i32;
             if ((bits & 0x7f800000U) == 0x7f800000U) {
-                const char* sign = (bits & 0x80000000U) ? "-" : "";
+                bool is_negative = (bits & 0x80000000U);
```

Comment:
Fix (and follow naming scheme):

```suggestion
                bool isNegative = (bits & 0x80000000U) != 0;
```

Reply:
Huh. These should be equivalent, no? This is what they both look like on Godbolt for me: https://godbolt.org/z/7hdqGoWM5

I will fix the case later though.

Reply:
I had a look into this and learned something new.

The Godbolt code isn't equivalent to the code in the project: 
- The code is using C++ mode, i.e. uses C++'s `bool` type
- Switching to C, it'll use C99's `bool` type instead. However, w2c2 is C89/90 compatible, so defines its own `bool` type for backward-compatibility

It turns out that for the built-in C99 `bool`, any non-zero value is converted to true.
However, w2c2's `bool` is defined as an enum, https://github.com/turbolent/w2c2/blob/main/w2c2/w2c2_base.h#L13-L16, so it may have a smaller size than 64-bit, in which case the conversion _truncates_ the higher-order bits (see https://godbolt.org/z/h6d365rTM).

See e.g. https://news.ycombinator.com/item?id=6309008

Reply:
Oh, no, didn't know you made your own `bool`. Good to know...